rust/core-lib/src/editor.rs

```diff
@@ -438,6 +438,7 @@ impl Editor {
         let keep_selections = self.this_edit_type == EditType::Transpose;
         self.scroll_to = self.view.after_edit(&self.text, &last_text, &delta, is_pristine, keep_selections);
         let (iv, new_len) = delta.summary();
+        let new_line_count = self.text.measure::<LinesMetric>() + 1;
```

Comment:
This may be a naming problem? this is supposed to represent "new count of total lines". `LinesMetric` returns 0 if there are no newline characters in the rope, although in this case the number of lines is still 1.